src/zcash/IncrementalMerkleTree.cpp

```diff
@@ -51,6 +437,410 @@ SHA256Compress SHA256Compress::combine(
+    }),
+    uint256(std::vector<unsigned char>{
+        0x88, 0x95, 0x2e, 0x3d, 0x0a, 0xc0, 0x6c, 0xb1,
+        0x6b, 0x66, 0x52, 0x01, 0x12, 0x22, 0x49, 0x65,
+        0x9a, 0x22, 0x32, 0x5e, 0x01, 0xc8, 0x70, 0xf4,
+        0x9e, 0x29, 0xda, 0x6b, 0x17, 0x57, 0xe0, 0x82,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xcd, 0xf8, 0x79, 0xf2, 0x43, 0x5b, 0x95, 0xaf,
+        0x04, 0x2a, 0x3b, 0xf7, 0xb8, 0x50, 0xf7, 0x81,
+        0x92, 0x46, 0xc8, 0x05, 0x28, 0x58, 0x03, 0xd6,
+        0x7f, 0xfb, 0xf4, 0xf2, 0x95, 0xbe, 0xd0, 0x04,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xe0, 0x05, 0xe3, 0x24, 0x20, 0x0b, 0x4f, 0x42,
+        0x8c, 0x62, 0xbc, 0x33, 0x31, 0xe6, 0x95, 0xc3,
+        0x73, 0x60, 0x7c, 0xd0, 0xfa, 0xa9, 0x79, 0x03,
+        0x41, 0xfa, 0x3b, 0xa1, 0xed, 0x22, 0x8b, 0xc5,
+    }),
+    uint256(std::vector<unsigned char>{
+        0x35, 0x44, 0x47, 0x72, 0x7a, 0xa9, 0xa5, 0x3d,
+        0xd8, 0x34, 0x5b, 0x6b, 0x6c, 0x69, 0x34, 0x43,
+        0xe5, 0x6e, 0xf4, 0xae, 0xba, 0x13, 0xc4, 0x10,
+        0x17, 0x9f, 0xc8, 0x58, 0x9e, 0x77, 0x33, 0xd5,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xda, 0x52, 0xdd, 0xa9, 0x1f, 0x28, 0x29, 0xc1,
+        0x5c, 0x0e, 0x58, 0xd2, 0x9a, 0x95, 0x36, 0x0b,
+        0x86, 0xab, 0x30, 0xcf, 0x0c, 0xac, 0x81, 0x01,
+        0x83, 0x2a, 0x29, 0xf3, 0x8c, 0x31, 0x85, 0xf1,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xc7, 0xda, 0x78, 0x14, 0xe2, 0x28, 0xe1, 0x14,
+        0x44, 0x11, 0xd7, 0x8b, 0x53, 0x60, 0x92, 0xfe,
+        0x92, 0x0b, 0xcd, 0xfc, 0xc3, 0x6c, 0xf1, 0x9d,
+        0x12, 0x59, 0x04, 0x7b, 0x26, 0x7d, 0x58, 0xb5,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xab, 0xa1, 0xf6, 0x8b, 0x6c, 0x2b, 0x4d, 0xb6,
+        0xcc, 0x06, 0xa7, 0x34, 0x0e, 0x12, 0x31, 0x3c,
+        0x4b, 0x4a, 0x4e, 0xa6, 0xde, 0xb1, 0x7d, 0xeb,
+        0x3e, 0x1e, 0x66, 0xcd, 0x8e, 0xac, 0xf3, 0x2b,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xc1, 0x60, 0xae, 0x4f, 0x64, 0xab, 0x76, 0x4d,
+        0x86, 0x4a, 0x52, 0xad, 0x5e, 0x33, 0x12, 0x6c,
+        0x4b, 0x5c, 0xe1, 0x05, 0xa4, 0x7d, 0xee, 0xdd,
+        0x75, 0xbc, 0x70, 0x19, 0x9a, 0x52, 0x47, 0xef,
+    }),
+    uint256(std::vector<unsigned char>{
+        0xea, 0xdf, 0x23, 0xfc, 0x99, 0xd5, 0x14, 0xdd,
+        0x8e, 0xa2, 0x04, 0xd2, 0x23, 0xe9, 0x8d, 0xa9,
+        0x88, 0x83, 0x1f, 0x9b, 0x5d, 0x19, 0x40, 0x27,
+        0x4c, 0xa5, 0x20, 0xb7, 0xfb, 0x17, 0x3d, 0x8a,
+    }),
+    uint256(std::vector<unsigned char>{
+        0x5b, 0x8e, 0x14, 0xfa, 0xca, 0xc8, 0xa7, 0xc7,
+        0xa3, 0xbf, 0xee, 0x8b, 0xae, 0x71, 0xf2, 0xf7,
+        0x79, 0x3d, 0x3a, 0xd5, 0xfe, 0x33, 0x83, 0xf9,
+        0x3a, 0xb6, 0x06, 0x1f, 0x2a, 0x11, 0xbb, 0x02
+    }),
+};
+
+SHA256Compress SHA256Compress::EmptyRoot(size_t depth) {
+    assert(depth <= sizeof(sha256_empty_roots)/sizeof(sha256_empty_roots[0]));
+    return sha256_empty_roots[depth];
```

Comment:
This alters the current behaviour. In the previous implementation `array::at` was used, resulting in an error being thrown if the index was out of bounds. Here we assert and then do an unchecked lookup, meaning that if assertions are disabled by someone, the out-of-bounds read will go through. Please replace these two lines with:
```suggestion
    return sha256_empty_roots.at(depth);
```